Tests/SwiftParserTest/Statements.swift

```diff
@@ -150,57 +150,54 @@ final class StatementTests: XCTestCase {
   func testAttributesOnStatements() {
     AssertParse(
       """
-      func test1() {#^END_FUNCTION^#
-        #^EXTRANEOUS^#@s return
+      func test1() {
+        #^TEST_1^#@s return
       }
       func test2() {
-        @unknown return
+        #^TEST_2^#@unknown return
       }
       """,
       diagnostics: [
-        DiagnosticSpec(locationMarker: "END_FUNCTION", message: "Expected '}' to end function"),
-        DiagnosticSpec(locationMarker: "EXTRANEOUS", message: "Extraneous code at top level")
+        DiagnosticSpec(locationMarker: "TEST_1", message: "Unexpected text '@s return' found in function"),
+        DiagnosticSpec(locationMarker: "TEST_2", message: "Unexpected text '@unknown return' found in function")
```

Comment:
`return` is a fairly strong keyword (IMO), should we really consume it as well?

Reply:
That would require more recovery logic in `parseStatement`. I filed https://github.com/apple/swift-syntax/issues/714 to do this later (after I have merged all my other recovery work).